source/Calamari.CloudAccounts/GoogleAuthenticationProvider.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Calamari.Common.Plumbing.Logging;
+using Calamari.Common.Plumbing.Variables;
+using Newtonsoft.Json;
+using NetWebRequest = System.Net.WebRequest;
+
+
+namespace Calamari.CloudAccounts
+{
+    public class GoogleAuthenticationProvider
+    {
+        readonly HttpClient httpClient;
+        const string StsEndpoint = "https://sts.googleapis.com/v1/token";
+        const string GcrScope = "https://www.googleapis.com/auth/devstorage.read_write";
+        const string OidcUsername = "oauth2accesstoken";
+        
+        public GoogleAuthenticationProvider()
+        {
+            httpClient = new HttpClient(new HttpClientHandler {Proxy = NetWebRequest.DefaultWebProxy});
```

Comment:
Probably not an issue, but I recall that directly constructing `HttpClient` can be problematic, depending on how frequently this class is instantiated?

Is it possible/a good idea to add the client as a constructor argument and inject a shared one in?

Reply:
From a quick search in the sln, it looks like this is a pattern we use in a few places already, so I think you can ignore this one!

Reply:
@wlthomson The execution model for calamari would mostly make HttpClient reuse redundant. The lifetime of the code is a single execution per step (Calamari is actually a CLI)